frontend/packages/dev-console/src/utils/useAddActionExtensions.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  ResolvedExtension,
+  useResolvedExtensions,
+  AddAction,
+  isAddAction,
+} from '@console/dynamic-plugin-sdk';
+
+interface AddPage {
+  disabledActions: string[];
+}
+
+export const useAddActionExtensions = (): [ResolvedExtension<AddAction>[], boolean] => {
+  const [allAddActionExtensions, resolved] = useResolvedExtensions<AddAction>(isAddAction);
+
+  if (window.SERVER_FLAGS.addPage) {
+    const addPage: AddPage = JSON.parse(window.SERVER_FLAGS.addPage);
```

Comment:
should this be in a try/catch block?
How about creating a separating hook which returns an empty array or undefined if no ids are present?

Reply:
I had a try catch here while developing, but then I decided to remove it because we could expect that if addPage is truthy (not null, not an empty string) that it could be parsed.

Extracting this into a hook (aka another function) is fine. I will do this.

Reply:
@sahil143 Added another hook `useDisabledAddActions` for extracting the SERVER_FLAGS.

Reply:
@jerolimov since `useDisabledAddActions` is function and not using any of the hooks. let's not use the hook naming convention for this.

Reply:
@sahil143 changed to `getDisabledAddActions`, can you check again?